### PR TITLE
Typo fixes and some readability tweaks

### PR DESCRIPTION
**Pull request by btamayo**
Improvements to docs, error message copy, and typo fixes – these were small improvements I added locally while I was trying to debug and understand some functionality and figured I might as well PR it.  

Signed-off-by: Bianca Tamayo <bianca@btamayo.me>

### CODE REVIEW

File: pkg/auth/auth.go [hunk 1]
```diff
@@ -29,13 +29,13 @@ import (
...
 // Authenticator is an interface to represent an external
-// source that should be authenticated against
+// source that should be authenticated against.
 type Authenticator interface {
 	// Auth is used to authenticate to an external service
 	Auth(*api.Client, cfg.Config) (string, error)
 }
 
-// authenticate authenticates!
+// authenticate authenticates with Vault.
```
Comment (broamski) [inline pkg/auth/auth.go:38]: Suggestion: "Authenticate authenticates to Vault, returns true if successful."

File: pkg/config/config.go [hunk 2]
```diff
@@ -10,7 +10,7 @@ import (
 
 const authPathFmtString = "auth/%s/login"
 
-//Config represents an application configurations
+// Config represents an application configurations
```
Comment (broamski) [inline pkg/config/config.go:13]: Suggestion: s/configurations/configuration/

**Review by broamski — APPROVED** [review]
Thanks for the improvements! I've left 2 small suggestions if you wouldn't mind including them.

File: pkg/config/config.go (hunk 2 above)
Comment (btamayo) [reply pkg/config/config.go]: Done! (I'll also likely be doing more of these types of changes as part of future any PRs.)

File: pkg/auth/auth.go (hunk 1 above)
Comment (btamayo) [reply pkg/auth/auth.go]: Done!

**Pull request merged**